test/services/videoconference/index.test.js

```diff
@@ -0,0 +1,219 @@
+		),
+		'feature should not be enabled in school')
+			.to.throw;
+		testData.school.features.push(VIDEOCONFERENCE);
+		await testData.school.save();
+		const response = await createService.create(testData.serviceParams, testData.teacherRequestAuthentication);
+		expect(response).to.be.ok;
+		expect(response.status).to.be.equal('SUCCESS');
+		expect(response.url).to.be.not.empty;
+	});
+
+	// it('test join with start permission works multiple times [slow]', async () => {
+	// 	let successfulRuns = 0;
+	// 	const authenticated = [];
+	// 	for (let i = 0; i < 20; i += 1) {
+	// 		const response = await createService.create(testData.serviceParams, testData.teacherRequestAuthentication);
+	// 		expect(response).to.be.ok;
+	// 		expect(response.status).to.be.equal('SUCCESS');
+	// 		expect(response.url).to.be.not.empty;
+	// 		authenticated.push(rp(response.url));
+	// 		successfulRuns += 1;
+	// 	}
+	// 	expect(successfulRuns).to.be.equal(20);
+	// 	// expect all requests finish successfully
+	// 	return Promise.all(authenticated);
+	// });
+
+	it('test join permission works for students and substitution teachers', async () => {
+		const studentResponse = await createService
+			.create(testData.serviceParams, testData.studentRequestAuthentication);
+		expect(studentResponse.url,
+			'students should not have the permission to join a started videoconference in courses')
+			.not.to.be.empty;
+
+		const substitutionteacherResponse = await createService
+			.create(testData.serviceParams, testData.substitutionTeacherRequestAuthentication);
+		expect(substitutionteacherResponse.url,
+			'substitutionteachers should be able to start a videoconference in courses')
+			.not.to.be.empty;
+	});
+
+	it('test creation without permission fails', () => {
+		expect(() => createService
+			.create(testData.serviceParams), 'no authentication should fail')
+			.to.throw;
+		expect(() => createService
+			.create(testData.serviceParams, testData.someUserAuth), 'missing scope permission should fail')
+			.to.throw;
+	});
+
+	it('test get with start or join permission works after creation', async () => {
+		// expect creation finished like above...
+		const teacherResponse = getService
+			.get(testData.serviceParams.scopeId, {
+				route: { scopeName: testData.serviceParams.scopeName },
+				...testData.teacherRequestAuthentication,
+			});
+		const substitutionTeacherResponse = getService
+			.get(testData.serviceParams.scopeId, {
+				route: { scopeName: testData.serviceParams.scopeName },
+				...testData.substitutionTeacherRequestAuthentication,
+			});
+		const courseStudentResponse = getService
+			.get(testData.serviceParams.scopeId, {
+				route: { scopeName: testData.serviceParams.scopeName },
+				...testData.studentRequestAuthentication,
+			});
+		let successCounter = 0;
+		await Promise.all([teacherResponse, substitutionTeacherResponse, courseStudentResponse])
+			.then(([...serviceResponses]) => Promise.all(serviceResponses.map((serviceResponse) => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected to return a value in arrow function.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4819756)